###### About Data
- For our project, we choose dataset from `TripAdvisor` having restaurant data across 7 major states in USA from [`kaggle`](https://www.kaggle.com/datasets/siddharthmandgi/tripadvisor-restaurant-recommendation-data-usa).
- This dataset was scraped from the TripAdvisor website and it conatins restaurant data across 20 cities in Washington, Texas, California, New York, Pensi, New Jersey and Oregon and Pennsylvania.

*Note:* Our dataset file can be accessible here on [github](https://github.com/sifat-e-noor/Recommendation-system).

###### Format and Clean the Dataset

For our further analysis to buil a resturant recommendation system, we clean and format the data set as per requirement. 

To find out issues with our choosen dataset, we will do the followings-
- Check for missing values for overall dataset
- Check closely for subtle values in other coulmn(s) through unique values except missing values column(s)
- Check closely for non-relatable strings except missing values column(s)
- Drop NaN values from required row(s)/column(s) 
- Fix unrelatable data values 
- Drop non require row(s)/column(s)
- Chnage data type (if require)
- Reorder column(s) and reset index (if require)

##### All important import(s) goes here

In [1]:
import os
import numpy as np
import pandas as pd
import requests

###### Download dataset (.csv) file from github

In [2]:
# Functin to download file from github
def file_exist(file_name):
    return file_name in os.listdir(os.getcwd())
def download_file(url,file_name):
    if file_exist(file_name):
        return
    response = requests.get(url)
    open(file_name, "wb").write(response.content)

In [3]:
# Load the dataset from git
url = "https://github.com/sifat-e-noor/Recommendation-system/raw/main/Data/TripAdvisor_RestauarantRecommendation.csv"
download_file(url,"TripAdvisor_RestauarantRecommendation.csv")
df_dataset = pd.read_csv("TripAdvisor_RestauarantRecommendation.csv")

# View the dataset's first 3 data
print(f"Our dataset's first 3 data:")
df_dataset.head(3)
# View full dataset
# df_dataset

Our dataset's first 3 data:


Name       Street Address  \
0  Betty Lou's Seafood and Grill     318 Columbus Ave   
1              Coach House Diner        55 State Rt 4   
2               Table Talk Diner  2521 South Rd Ste C   

                       Location                                          Type  \
0  San Francisco, CA 94133-3908   Seafood, Vegetarian Friendly, Vegan Options   
1     Hackensack, NJ 07601-6337          Diner, American, Vegetarian Friendly   
2   Poughkeepsie, NY 12601-5476          American, Diner, Vegetarian Friendly   

            Reviews No of Reviews  \
0  4.5 of 5 bubbles   243 reviews   
1    4 of 5 bubbles    84 reviews   
2    4 of 5 bubbles   256 reviews   

                                            Comments   Contact Number  \
0                                                NaN  +1 415-757-0569   
1  Both times we were there very late, after 11 P...  +1 201-488-4999   
2  Waitress was very friendly but a little pricey...  +1 845-849-2839   

                                    Trip_advisor Url  \
0  https://www.tripadvisor.com//Restaurant_Review...   
1  https://www.tripadvisor.com//Restaurant_Review...   
2  https://www.tripadvisor.com//Restaurant_Review...   

                                        Menu Price_Range  
0               Check The Website for a Menu    $$ - $$$  
1               Check The Website for a Menu    $$ - $$$  
2  http://tabletalkdiner.com/menu/breakfast/    $$ - $$$

In [4]:
# Obtain basic info of dataset (i.e shape, row and column) 
print("Shape of dataset", df_dataset.shape)
print("Size of dataset", df_dataset.size)
print(f"It has {len(df_dataset.axes[0])} rows, and {len(df_dataset.axes[1])} columns titled ase:")
for i, col in enumerate(df_dataset.axes[1]):
    print(i+1, col)

Shape of dataset (3062, 11)
Size of dataset 33682
It has 3062 rows, and 11 columns titled ase:
1 Name
2 Street Address
3 Location
4 Type
5 Reviews
6 No of Reviews
7 Comments
8 Contact Number
9 Trip_advisor Url
10 Menu
11 Price_Range


- From the above observation, we found that our dataset has the following attriutes:

__1.__ Name (Name of restuarants), __2.__ Streest Address (Name of the Street), __3.__ Location (City and Zip code), __4.__ Type (Types of cuisines served), __5.__ Reviews (Star Ratings), __6.__ No of Reviews (No. of People who have rated), __7.__ Comments (Customer reviews), __8.__ Contact Number (USA - Phone Number), __9.__ Trip_advisor Url, __10.__ Menu URL, and __11.__ Price_Range. 

- For our further analysis, We dont need columns from 8 - 10 as these columns values do not contribute in our goal's analysis(i.e 8. Contact Number, 9. Menu url, and 10. Trip Advisor Url). Therefore, to make the dataset more viusally manageable within the screensize, we discard these columns.

In [5]:
# Drop columns form 8-10(i.e 8. Contact Number, 9 Trip_advisor Url, 10 Menu) 
df_new_dataset = df_dataset.drop(df_dataset.loc[:, 'Contact Number':'Menu'].columns, axis=1)

# Rename 2 columns to check those columns' values
df_new_dataset.rename(columns = {'Street Address':'Street_Address', 'No of Reviews':'No_of_Reviews'}, inplace = True)

# View the dataset's first 3 data
print(f"Our dataset's first 3 data after dropping out 3 columns:")
# df_new_dataset.head(3)
df_new_dataset

Our dataset's first 3 data after dropping out 3 columns:


Name                         Street_Address  \
0     Betty Lou's Seafood and Grill                       318 Columbus Ave   
1                 Coach House Diner                          55 State Rt 4   
2                  Table Talk Diner                    2521 South Rd Ste C   
3                       Sixty Vines                       3701 Dallas Pkwy   
4                      The Clam Bar                      3914 Brewerton Rd   
...                             ...                                    ...   
3057     Indigo Kitchen & Ale House                       2902 164th St SW   
3058                  City Vineyard                        233 West Street   
3059             BRIO Tuscan Grille                      3710 US Highway 9   
3060          Maywood Pancake house                      92 W Pleasant Ave   
3061                  Porto Leggero  Harborside Financial Center - Plaza 5   

                          Location  \
0     San Francisco, CA 94133-3908   
1        Hackensack, NJ 07601-6337   
2      Poughkeepsie, NY 12601-5476   
3             Plano, TX 75093-7777   
4               Syracuse, NY 13212   
...                            ...   
3057       Lynnwood, WA 98087-3201   
3058       New York City, NY 10013   
3059       Freehold, NJ 07728-4801   
3060        Maywood, NJ 07607-1336   
3061         Jersey City, NJ 07311   

                                               Type           Reviews  \
0       Seafood, Vegetarian Friendly, Vegan Options  4.5 of 5 bubbles   
1              Diner, American, Vegetarian Friendly    4 of 5 bubbles   
2              American, Diner, Vegetarian Friendly    4 of 5 bubbles   
3           American, Wine Bar, Vegetarian Friendly  4.5 of 5 bubbles   
4                            American, Bar, Seafood    4 of 5 bubbles   
...                                             ...               ...   
3057             American, Bar, Vegetarian Friendly  4.5 of 5 bubbles   
3058                         American, Bar, Seafood  4.5 of 5 bubbles   
3059                       Steakhouse, Italian, Bar    4 of 5 bubbles   
3060   American, Vegan Options, Gluten Free Options    4 of 5 bubbles   
3061    Italian, Vegetarian Friendly, Vegan Options  4.5 of 5 bubbles   

     No_of_Reviews                                           Comments  \
0      243 reviews                                                NaN   
1       84 reviews  Both times we were there very late, after 11 P...   
2      256 reviews  Waitress was very friendly but a little pricey...   
3      235 reviews  Not sure why I went there for the second time....   
4      285 reviews  Doesn't look like much from the outside but wa...   
...            ...                                                ...   
3057   198 reviews  We had to wait a few minutes to get it but it ...   
3058   374 reviews  Came here to see Andrea Gibson perform, we wer...   
3059   211 reviews  We come to Brio once a month. They are good fo...   
3060    87 reviews  Stopped in yesterday for Breakfast. When in Je...   
3061   183 reviews  First time for dinner. 2 couples. Delicious mu...   

     Price_Range  
0       $$ - $$$  
1       $$ - $$$  
2       $$ - $$$  
3       $$ - $$$  
4       $$ - $$$  
...          ...  
3057    $$ - $$$  
3058    $$ - $$$  
3059    $$ - $$$  
3060    $$ - $$$  
3061        $$$$  

[3062 rows x 8 columns]

- To find issues (i.e. missing values) in our data set, we get dataset's basic info first.

In [6]:
# View dataset's per columns overall basic info
print('Basic info of dataset:\n') 
df_new_dataset.info()
print('\n')
# Find dataset's overall missing values per columns
print('Follwing Columns have missing values:')
df_new_dataset.isnull().sum()

Basic info of dataset:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3062 entries, 0 to 3061
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Name            3062 non-null   object
 1   Street_Address  3062 non-null   object
 2   Location        3062 non-null   object
 3   Type            3049 non-null   object
 4   Reviews         3062 non-null   object
 5   No_of_Reviews   3062 non-null   object
 6   Comments        2447 non-null   object
 7   Price_Range     3062 non-null   object
dtypes: object(8)
memory usage: 191.5+ KB


Follwing Columns have missing values:


Name                0
Street_Address      0
Location            0
Type               13
Reviews             0
No_of_Reviews       0
Comments          615
Price_Range         0
dtype: int64

- Check closely for subtle values

In [7]:
# Find unique values for each Name column to see there is any non-meaningful string exist
print(f'Name column has: {len(df_new_dataset.Name.unique())} data')
print("Unique values of "'Name'" column:")
df_new_dataset.Name.unique()

Name column has: 2641 data
Unique values of Name column:


array(["Betty Lou's Seafood and Grill", 'Coach House Diner',
       'Table Talk Diner', ..., 'BRIO Tuscan Grille',
       'Maywood Pancake house', 'Porto Leggero'], dtype=object)

In [8]:
# As Name column has 3062 data, we can't review all unique values in bare eyes. 
# Therefore, we check closely Name column to find unrelateable string data
is_no_name = False
for name in df_new_dataset.Name:    
    if name == 'No name' and name == 'Undefined':
        is_no_name = True
        print(name)
    else:
        pass
if is_no_name == True:
    print("Unrelatable data exists")
else:
    print("No unrelateable data exist in Name column")

No unrelateable data exist in Name column


In [9]:
# Find unique values for Street_Address column to see there is any non-meaningful string exist
print(f'Street_Address column has: {len(df_new_dataset.Street_Address.unique())} data')
print("Unique values of "'Street_Address'" column:")
df_new_dataset.Name.unique()

Street_Address column has: 2814 data
Unique values of Street_Address column:


array(["Betty Lou's Seafood and Grill", 'Coach House Diner',
       'Table Talk Diner', ..., 'BRIO Tuscan Grille',
       'Maywood Pancake house', 'Porto Leggero'], dtype=object)

In [10]:
# As Street_Address column has 2813 data, we can't review all unique values in bare eyes. 
# Therefore, we check closely Street_Address column to find unrelateable string data
is_no_address = False
for address in df_new_dataset.Street_Address:    
    if address == 'No address' and address == 'Undefined':
        is_no_address = True
        print(address)
    else:
        pass
if is_no_address == True:
    print("Unrelatable data exists")
else:
    print("No unrelateable data exists in Street_Address column")

No unrelateable data exists in Street_Address column


In [11]:
# Find unique values for Location column to see there is any non-meaningful string exist
print(f'Location column has: {len(df_new_dataset.Location.unique())} data')
print("Unique values of "'Location'" column:")
df_new_dataset.Location.unique()

Location column has: 2653 data
Unique values of Location column:


array(['San Francisco, CA 94133-3908', 'Hackensack, NJ 07601-6337',
       'Poughkeepsie, NY 12601-5476', ..., 'Freehold, NJ 07728-4801',
       'Maywood, NJ 07607-1336', 'Jersey City, NJ 07311'], dtype=object)

In [12]:
# As Location column has 3062 data, we can't review all unique values in bare eyes. 
# Therefore, we check closely Location column to find unrelateable string data
is_no_location = False
for location in df_new_dataset.Location:    
    if location == 'No location' and location == 'Undefined':
        is_no_location = True
        print(location)
    else:
        pass
if is_no_location == True:
    print("Unrelatable data exists")
else:
    print("No unrelateable data exist in Location column")

No unrelateable data exist in Location column


In [13]:
# Find unique values for Reviews column to see there is any non-meaningful string exist
print(f'Reviews column has: {len(df_new_dataset.Reviews.unique())} data')
print("Unique values of "'Reviews'" column:")
df_new_dataset.Reviews.unique()

Reviews column has: 6 data
Unique values of Reviews column:


array(['4.5 of 5 bubbles', '4 of 5 bubbles', '5 of 5 bubbles',
       '3.5 of 5 bubbles', '3 of 5 bubbles', 'No review'], dtype=object)

In [14]:
# To increase readability, we remove redundent words from Reviews column 
df_new_dataset['Reviews'] = [n.split()[0] for n in df_new_dataset['Reviews']]

In [15]:
# Find unique values for No_of_Reviews column 
print(f'No_of_Reviews column has:{len(df_new_dataset.No_of_Reviews.unique())} data')
print("Unique values of "'No_of_Reviews'" column:")
print(df_new_dataset.No_of_Reviews.unique())

No_of_Reviews column has:811 data
Unique values of No_of_Reviews column:
['243 reviews' '84 reviews' '256 reviews' '235 reviews' '285 reviews'
 '1,406 reviews' '220 reviews' '89 reviews' '90 reviews' '258 reviews'
 '271 reviews' '87 reviews' '118 reviews' '706 reviews' '104 reviews'
 '314 reviews' '193 reviews' '1,198 reviews' '280 reviews' '1,807 reviews'
 '642 reviews' '928 reviews' '30 reviews' '179 reviews' '746 reviews'
 '119 reviews' '829 reviews' '515 reviews' '123 reviews' '189 reviews'
 '39 reviews' '742 reviews' '264 reviews' '399 reviews' '111 reviews'
 '24 reviews' '187 reviews' '1,653 reviews' '223 reviews' '153 reviews'
 '240 reviews' '364 reviews' '191 reviews' '775 reviews' '124 reviews'
 '198 reviews' '292 reviews' '29 reviews' '683 reviews' '27 reviews'
 '214 reviews' '513 reviews' '44 reviews' '28 reviews' '611 reviews'
 '307 reviews' '440 reviews' '660 reviews' '413 reviews' '135 reviews'
 '74 reviews' '160 reviews' '273 reviews' '86 reviews' '932 reviews'
 '378 rev

In [16]:
# To increase readability, we remove redundent words from No_of_Reviews column
df_new_dataset['No_of_Reviews'] = [n.split()[0].replace(',', '') for n in df_new_dataset['No_of_Reviews']]

In [17]:
# As No_of_Reviews column has 810 data, we can't review all unique values in bare eyes. 
# Therefore, we check closely No_of_Reviews column to find unrelateable string data
is_no_of_reviews = False
for review in df_new_dataset.No_of_Reviews:    
    if review == 'No review' or review == 'Undefined':
        is_no_of_reviews = True
        print(review)
    else:
        pass
if is_no_of_reviews:
    print("Unrelatable data exists")
else:
    print("No unrelateable data exist in No_of_Reviews column")    

Undefined
Undefined
Unrelatable data exists


In [18]:
# Find unique values for Price_Range column to see there is any non-meaningful string exist
print(f'Price_Range column has: {len(df_new_dataset.Price_Range.unique())} data')
print("Unique values of "'Price_Range'" column:")
df_new_dataset.Price_Range.unique()

Price_Range column has: 6 data
Unique values of Price_Range column:


array(['$$ - $$$', '$$$$', '$', 'American', 'Not Specified', 'Italian'],
      dtype=object)

- Assign descriptive values for price range to make the data more readable

In [19]:
# Format Price_Range column's value
def price_range(price_value):
    if price_value == '$$ - $$$':
        return  "Medium"
    elif price_value == '$$$$':
        return  "Expensive"
    else:
        return  "Cheap"    

df_new_dataset['Price_Range'] = df_new_dataset['Price_Range'].apply(price_range)

In [20]:
df_new_dataset.Price_Range.unique()

array(['Medium', 'Expensive', 'Cheap'], dtype=object)

In [21]:
# Find unique values for Type column to see there is any non-meaningful string exist
print(f'Type column has: {len(df_new_dataset.Type.unique())} data')
print("Unique values of "'Type'" column:")
df_new_dataset.Type.unique()

Type column has: 670 data
Unique values of Type column:


array([' Seafood, Vegetarian Friendly, Vegan Options',
       ' Diner, American, Vegetarian Friendly',
       ' American, Diner, Vegetarian Friendly',
       ' American, Wine Bar, Vegetarian Friendly',
       ' American, Bar, Seafood', ' Seafood, Italian, Pizza',
       ' American, Steakhouse, Bar', ' Mediterranean, Greek, Healthy',
       ' Italian, Pizza, Vegetarian Friendly',
       ' American, Steakhouse, Gluten Free Options',
       ' Italian, Vegetarian Friendly, Gluten Free Options',
       ' Cafe, American, Vegetarian Friendly', ' Irish, Bar, Pub',
       ' American, Bar, Vegetarian Friendly', ' American, Cafe, Healthy',
       ' American, Bar, Pub', ' Steakhouse, Latin, Argentinean',
       ' Chinese, Asian, Taiwanese', ' Belgian, French, Cafe',
       ' American, Cajun & Creole, Vegetarian Friendly',
       ' Steakhouse, Mexican, Vegetarian Friendly',
       ' American, Vegetarian Friendly, Vegan Options',
       ' Bar, Contemporary, Fusion',
       ' American, Bar, Gluten Fr

In [22]:
# As Type column has 670 data, we can't review all unique values in bare eyes. 
# Therefore, we check closely Comments column to find unrelateable string data
is_type = False
for types in df_new_dataset.Type:    
    if types == 'No type' or types == 'Undefined':
        is_type = True
        print(types)
    else:
        pass
if is_type:
    print("Unrelatable data exists")
else:
    print("No unrelateable data exist in Type column")  

No unrelateable data exist in Type column


In [23]:
# As Comments column has 2447 data, we can't review all unique values in bare eyes. 
# Therefore, we check closely Comments column to find unrelateable string data
is_comments = False
for comment in df_new_dataset.Comments:    
    if comment == 'No comment' or comment == 'Undefined':
        is_comments = True
        print(comment)
    else:
        pass
if is_comments:
    print("Unrelatable data exists")
else:
    print("No unrelateable data exist in Comments column")  

No unrelateable data exist in Comments column


- As we will work with states of USA, so we will derive `state`, `city` and `Zip_Code`  from `location` column.

In [24]:
# Derive City from Location
def city_location(location):
    return location.split(',')[0].strip()
df_new_dataset['City'] = df_new_dataset['Location'].apply(city_location)

In [25]:
print(f'Total no. of cities: {len(df_new_dataset.City.unique())}')
print(f'All cities: {df_new_dataset.City.unique()}')

Total no. of cities: 200
All cities: ['San Francisco' 'Hackensack' 'Poughkeepsie' 'Plano' 'Syracuse'
 'Federal Way' 'Dallas' 'Renton' 'Ithaca' 'Vancouver' 'Kirkland' 'Albany'
 'Montclair' 'Princeton' 'Pasadena' 'Bellevue' 'Saratoga Springs'
 'Anaheim' 'Frisco' 'Atlantic City' 'Bothell' 'San Diego' 'Seattle'
 'Teaneck' 'Santa Monica' 'Staten Island' 'Toms River' 'Laredo'
 'Bakersfield' 'Los Angeles' 'Amarillo' 'Long Beach' 'New York City'
 'Freehold' 'Austin' 'Fresno' 'Berkeley' 'Yonkers' 'San Jose' 'Fort Worth'
 'Riverside' 'Newport Beach' 'Waco' 'Huntington Beach' 'Kennewick'
 'Ship Bottom' 'El Paso' 'Forest Hills' 'Puyallup' 'Sacramento'
 'Bellingham' 'Tustin' 'Spokane Valley' 'Kent' 'Cherry Hill'
 'New Brunswick' 'Katy' 'Everett' 'The Woodlands' 'Palm Springs'
 'Cedar Grove' 'Elizabeth' 'South Ozone Park' 'Newark' 'Glendale'
 'Hermosa Beach' 'Costa Mesa' 'Richardson' 'Lubbock' 'Redmond'
 'Ocean City' 'Niagara Falls' 'Yakima' 'Brownsville' 'Corpus Christi'
 'Irving' 'Houston' 'Spring

In [26]:
# Derive State from Location
def state_location(location):
        if len(location.split(',')) == 2:
            return location.split(',')[1][:3].strip() 
        else:
            return location.split(',')[2][:3].strip() 
        
df_new_dataset['State'] = df_new_dataset['Location'].apply(state_location)

In [27]:
print(f'Total no. of states: {len(df_new_dataset.State.unique())}')
print(f'All states: {df_new_dataset.State.unique()}')

Total no. of states: 8
All states: ['CA' 'NJ' 'NY' 'TX' 'WA' 'OR' 'PA' '']


- `CA` (California) state droped out from the dataset as we droped `NaN` values from `Type` and `comments` Columns. We have five different states of USA named `New Jersy`, `New York`, `Texas`, `Washington`, `Oregon` and `Pennsylvania`. 

In [28]:
# Ensure the CA state's resturants have no comments from users 
# df_new_dataset[df_new_dataset["State"]=="CA"][["Name","State","Comments"]].to_csv("State_CA_Resturants_Comments.csv")

In [29]:
# Find dataset's overall missing values per columns
print('Follwing Columns have missing values:')
df_new_dataset.isnull().sum()

Follwing Columns have missing values:


Name                0
Street_Address      0
Location            0
Type               13
Reviews             0
No_of_Reviews       0
Comments          615
Price_Range         0
City                0
State               0
dtype: int64

- From above table we can see that `Type` and `comments` have 13 and 615 visible missing values. Lets drop these two columns NaN values

In [30]:
# Drop NaN values
df_new_dataset.dropna(inplace=True, axis=0)
df_new_dataset.isnull().sum()

Name              0
Street_Address    0
Location          0
Type              0
Reviews           0
No_of_Reviews     0
Comments          0
Price_Range       0
City              0
State             0
dtype: int64

In [31]:
# Check empty string in State column 
df_new_dataset[df_new_dataset['State'] == '']

Name       Street_Address                          Location  \
2617  The Watermark  6361 Fallsview Blvd  Niagara Falls, Ontario L2G 3V9,    

                                   Type Reviews No_of_Reviews  \
2617   American, Canadian, Contemporary       4          1342   

                                               Comments Price_Range  \
2617  Our dining experience at The Watermark this pa...   Expensive   

               City State  
2617  Niagara Falls

In [32]:
# Drop empty string in State column
df_new_dataset.drop(df_new_dataset[(df_new_dataset['State'] == '')].index, inplace=True)

- We checked empty string as we found its existance deriving from State column's unique value. We droped the row that contain `province of Canada`(i.e. empty string in State column) as we will work with states of USA. 

In [33]:
# Derive Zip_Code from Location
def zipcode_location(zipcode):
    return zipcode.split(",")[-1].split(" ")[-1]
        
df_new_dataset['Zip_Code'] = df_new_dataset['Location'].apply(zipcode_location)

In [34]:
# Drop Location column
df_new_dataset = df_new_dataset.drop(['Location'], axis=1)

We also droped the `Location` column as we got our desired column `City`, `State` and `Zip_Code` from it.

In [35]:
# Dataset info droping all NaN values and non required row(s) and column(s)
df_new_dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2439 entries, 1 to 3061
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Name            2439 non-null   object
 1   Street_Address  2439 non-null   object
 2   Type            2439 non-null   object
 3   Reviews         2439 non-null   object
 4   No_of_Reviews   2439 non-null   object
 5   Comments        2439 non-null   object
 6   Price_Range     2439 non-null   object
 7   City            2439 non-null   object
 8   State           2439 non-null   object
 9   Zip_Code        2439 non-null   object
dtypes: object(10)
memory usage: 209.6+ KB


- Convert `Review` column's data to `float` and `No_of_Reviews` column's data to `int`. And reorder the columns.

In [36]:
# Review column's data to float
df_new_dataset['Reviews'] = df_new_dataset['Reviews'].apply(float)

In [37]:
# Convert No_of_Reviews column's data to int
df_new_dataset['No_of_Reviews'] = df_new_dataset['No_of_Reviews'].apply(int)

In [38]:
# Reordering columns
df_new_dataset = df_new_dataset.loc[:,['Name', 'Street_Address', 'State', 'City', 'Zip_Code', 'Type', 'Reviews', 'No_of_Reviews', 'Comments', 'Price_Range']]

- Reset index as we dropped some columns and rows 

In [39]:
# Reset index of dataset
df_new_dataset.reset_index(drop=True, inplace=True)

# View dataset's overall basic info by per columns
print('Basic info of dataset:\n') 
df_new_dataset.info()

Basic info of dataset:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2439 entries, 0 to 2438
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Name            2439 non-null   object 
 1   Street_Address  2439 non-null   object 
 2   State           2439 non-null   object 
 3   City            2439 non-null   object 
 4   Zip_Code        2439 non-null   object 
 5   Type            2439 non-null   object 
 6   Reviews         2439 non-null   float64
 7   No_of_Reviews   2439 non-null   int64  
 8   Comments        2439 non-null   object 
 9   Price_Range     2439 non-null   object 
dtypes: float64(1), int64(1), object(8)
memory usage: 190.7+ KB


In [40]:
# Save data set to .csv format
df_new_dataset.to_csv('TripAdvisor_RestauarantRecommendation_cleandataset.csv')